src/sortStudents.ts

```diff
@@ -1,16 +1,59 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+export type SortFunc = (a: Student[], b: SortType, c: SortOrder) => Student[];
 
+export const sortStudents: SortFunc = (students, sortBy, order) => {
+  type Num = number;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
-}
+  const sortCallback = (a: Student, b: Student): Num => {
+    const studentA: Student = order === 'asc'
+      ? a
+      : b;
+
+    const studentB: Student = order === 'asc'
+      ? b
+      : a;
+
+    const getSum = (x: Num, y: Num): Num => x + y;
+    const getAverageGradeOf = (student: Student): Num => (
+      student.grades.reduce(getSum) / student.grades.length
+    );
+
+    switch (sortBy) {
+      case SortType.Name:
+        return studentA.name.localeCompare(studentB.name);
+
+      case SortType.Surname:
+        return studentA.surname.localeCompare(studentB.surname);
```

Comment:
these cases are essentially the same - let's join them
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return studentA.surname.localeCompare(studentB.surname);
```